digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficLightDao.scala

```diff
@@ -214,28 +251,80 @@ object OracleTrafficLightDao {
     (StaticQuery.query[Long, String](propertyTypeByPropertyId).apply(propertyId).first, Some(propertyId), property)
   }
 
-  def singleChoiceValueDoesNotExist(assetId: Long, propertyId: Long) = {
-    StaticQuery.query[(Long, Long), Long](existsSingleChoiceProperty).apply((assetId, propertyId)).firstOption.isEmpty
+  def singleChoiceValueDoesNotExist(assetId: Long, propertyId: Long, groupedId: Option[Long] = None) = {
+    StaticQuery.query[(Long, Long, Option[Long]), Long](existsSingleChoicePropertyWithGroupedId).apply((assetId, propertyId, groupedId)).firstOption.isEmpty
+  }
+
+  def textPropertyValueDoesNotExist(assetId: Long, propertyId: Long, groupedId: Option[Long] = None) = {
+    StaticQuery.query[(Long, Long, Option[Long]), Long](existsTextPropertyWithGroupedId).apply((assetId, propertyId, groupedId)).firstOption.isEmpty
   }
 
-  def textPropertyValueDoesNotExist(assetId: Long, propertyId: Long) = {
-    StaticQuery.query[(Long, Long), Long](existsTextProperty).apply((assetId, propertyId)).firstOption.isEmpty
+  def multipleChoiceValueDoesNotExist(assetId: Long, propertyId: Long, groupedId: Option[Long] = None): Boolean = {
+    StaticQuery.query[(Long, Long, Option[Long]), Long](existsMultipleChoicePropertyWithGroupedId).apply((assetId, propertyId, groupedId)).firstOption.isEmpty
   }
 
-  def multipleChoiceValueDoesNotExist(assetId: Long, propertyId: Long): Boolean = {
-    StaticQuery.query[(Long, Long), Long](existsMultipleChoiceProperty).apply((assetId, propertyId)).firstOption.isEmpty
+  private def numberPropertyValueDoesNotExist(assetId: Long, propertyId: Long, groupedId: Option[Long] = None) = {
+    StaticQuery.query[(Long, Long, Option[Long]), Long](existsNumberPropertyWithGroupedId).apply((assetId, propertyId, groupedId)).firstOption.isEmpty
   }
 
-  def createOrUpdateProperties(assetId: Long, propertyPublicId: String, propertyId: Long, propertyType: String, propertyValues: Seq[PointAssetValue]) {
+  private def existingGroupedIdForAsset(assetId: Long): Set[Long] = {
+    StaticQuery.query[Long, Long](existingGroupedIdForAssetQuery).apply(assetId).iterator.toSet
+  }
+
+  def createOrUpdateProperties(assetId: Long, propertyPublicId: String, propertyId: Long, propertyType: String, propertyValues: Seq[PointAssetValue], groupedId: Option[Long]) {
```

Comment:
Can we initialize one variable with the property value at the beginning of this method to avoid the train in all case options?

```Scala
val propertyValue = propertyValues.head.asInstanceOf[PropertyValue].propertyValue
```

And then we can use it to convert to all types.

Reply:
Yes we can. Suggestion implemented